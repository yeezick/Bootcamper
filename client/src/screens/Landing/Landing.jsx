import React from 'react'

export const Landing = () => {
  return (
    <div>
      Landing Page
    </div>
  )
}
