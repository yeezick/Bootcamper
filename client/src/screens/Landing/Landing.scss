@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap');

.landing-page {
    font-family: 'Roboto', sans-serif;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    .img-placeholder {
        width: 100%;
        background-color: black;
        height: 280px;
    }
    h1 {
        font-size: 24px;
        font-weight: 700px;
        margin: 20px 0 10px 0;
    }
    button {
        background-color: black;
        border-style: none;
        border-radius: 4px;
        color: white;
        padding: 10px 20px;
        margin: 9px;
        width: 90px;
        font-size: 12px;
    }
    h5 {
        margin-top: 15px;
    }
    h6 {
        margin-top: 25px;
        font-size: 12px;
    }
}