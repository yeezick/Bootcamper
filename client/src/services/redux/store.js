import { configureStore } from '@reduxjs/toolkit';
import { uiSlice } from './slices/uiSlice';
import { projectSlice } from './slices/projectSlice';
import { counterSlice } from './slices/counterSlice';

const store = configureStore({
  reducer: {
    ui: uiSlice.reducer,
    projects: projectSlice.reducer,
    counter: counterSlice.reducer,
  },
});

export default store;
