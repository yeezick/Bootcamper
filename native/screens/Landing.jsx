import { StyleSheet, Text, View } from 'react-native';
import { SingleActionButton } from '../components/Button/SingleActionButton.jsx';

export const Landing = () => {
  const routeToSignUp = {
    path: 'SignUp',
    text: 'Sign Up',
    type: 'reroute',
  };

  const routeToSignIn = {
    path: 'SignIn',
    text: 'Sign in',
    type: 'reroute',
  };

  const routeToRoulette = {
    path: 'Roulette',
    text: 'Try it!',
    type: 'reroute',
  };

  return (
    <View style={styles.container}>
      <View style={styles.placeholderImage}></View>
      <Text>Bootcamper</Text>
      <SingleActionButton payload={routeToSignUp} />
      <SingleActionButton payload={routeToSignIn} />
      <Text>Want to take a test drive first?</Text>
      <SingleActionButton payload={routeToRoulette} />
      <Text> Help | Contact</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
  },
  placeholderImage: {
    backgroundColor: 'black',
    height: '60%',
    width: '100%',
  },
});
