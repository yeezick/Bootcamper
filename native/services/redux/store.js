import { configureStore } from '@reduxjs/toolkit';
import { uiSlice } from './slices/uiSlice';
import { projectSlice } from './slices/projectSlice';
import { counterSlice } from './slices/counterSlice';
import { toolSlice } from './slices/toolSlice';

const store = configureStore({
  reducer: {
    ui: uiSlice.reducer,
    projects: projectSlice.reducer,
    counter: counterSlice.reducer,
    tools: toolSlice.reducer,
  },
});

export default store;
