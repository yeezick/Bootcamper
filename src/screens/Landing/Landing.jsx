import { useNavigate } from 'react-router-dom';
import './Landing.scss';
import { SingleActionButton } from '../../components/Button/SingleActionButton';

export const Landing = () => {
  const navigate = useNavigate();
  const routeToSignUp = {
    handler() {
      navigate('sign-up');
    },
    title: 'Sign Up',
  };

  const routeToSignIn = {
    handler() {
      navigate('sign-in');
    },
    title: 'Sign In',
  };

  return (
    <div className="landing-page">
      <div className="img-placeholder"></div>
      <h1 className="title">Bootcamper</h1>
      <h3 className="subtitle">Find your next project!</h3>
      <SingleActionButton payload={routeToSignUp} style="default" size="long" />
      <SingleActionButton payload={routeToSignIn} style="light" size="long" />
      <p className="text">Continue with</p>
      <div className="sso-wrapper">
        <div className="sso-item"></div>
        <div className="sso-item"></div>
        <div className="sso-item"></div>
      </div>
      <a className="trial-link" href="/roulette">
        Continue without logging in
      </a>
    </div>
  );
};
