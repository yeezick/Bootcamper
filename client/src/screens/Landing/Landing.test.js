/**
 ** Initial Thoughts:
 * What to test on this screen?
 * - Does the screen load successfully?
 * - Are there 2 buttons on this screen?
 *    - Does clicking on these buttons successfuly navigate me to the right screens?
 * - Does it have the header
 * - Do the buttons navigate correctly?
 *
 ** Key Takeaways:
 * - Not worth testing styling or if the component is rendered. One could argue each button's test expecting it to be in the document would also be useless.
 * - Unit test should ONLY be focused on actual logic
 * - Checking browser history is really unstable & not recommended (wasted 4 hours on this). To test successful navigation, each screen should import & test the existence of a `<LocationDisplay />`
 */

import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../services/redux/store';
import { Landing } from './Landing';

describe('On initial render, there should be an image and 4 buttons', () => {
  const user = userEvent.setup();
  const renderLandingScreen = () =>
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Landing />
        </BrowserRouter>
      </Provider>
    );

  // it('Renders', () => {
  //   const { asFragment } = render(
  //     <Provider store={store}>
  //       <BrowserRouter>
  //         <Landing />
  //       </BrowserRouter>
  //     </Provider>
  //   );
  //   expect(asFragment()).toMatchSnapshot();
  // });

  beforeEach(() => {
    renderLandingScreen();
  });

  it('Should have a placeholder image of 82x82 pixels', async () => {
    const landingImg = screen.queryByTestId(/landing-image/i);
    const style = window.getComputedStyle(landingImg);
    expect(landingImg).toBeInTheDocument();
  });

  it('Log in button exists & navigates to sign in screen', async () => {
    const loginButton = screen.queryByText(/log in/i);
    expect(loginButton).toBeInTheDocument();
    await user.click(loginButton);
    const formButton = screen.getByRole('button');
    expect(formButton).toBeInTheDocument();
    expect(signinButton.innerText).toBe(/sign in/i);
  });

  it('Sign up button exists & navigates to sign up screen', async () => {
    const signupButton = screen.queryByText(/sign up/i);
    expect(signupButton).toBeInTheDocument();
    await user.click(signupButton);
    const formButton = screen.getByRole('button');
    expect(formButton).toBeInTheDocument();
    expect(formButton.innerText).toBe(/sign in/i);
  });

  it('Continue without logging in button exists & navigates to roulette screen', async () => {
    const continueButton = screen.queryByText(/continue without logging in/i);
    expect(continueButton).toBeInTheDocument();
    await user.click(continueButton);
    const declineButton = screen.getByRole('roulette-decline');
    expect(declineButton).toBeInTheDocument();
    // what else to test if decline button exists
  });
});
