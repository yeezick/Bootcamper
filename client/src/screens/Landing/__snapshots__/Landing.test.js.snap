// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On initial render, there should be an image and 4 buttons Renders 1`] = `
<DocumentFragment>
  <div
    class="landing-page"
  >
    <div
      class="img-placeholder"
      data-testid="landing-image"
    />
    <h1
      class="logo-placeholder"
    >
      Bootcamper
    </h1>
    <div
      class="single-button"
    >
      <button
        type="button"
      >
        Sign Up
      </button>
    </div>
    <div
      class="single-button"
    >
      <button
        type="button"
      >
        Log In
      </button>
    </div>
    <h5>
      Want to take a test drive first?
    </h5>
    <div
      class="single-button"
    >
      <button
        type="button"
      >
        Try it!
      </button>
    </div>
    <h6>
      Help | Contact
    </h6>
  </div>
</DocumentFragment>
`;
